### ISSUE TITLE
unable to place stop loss on exchange when low amount of a coin has been buy (due to partially fullfilled)

### ISSUE THREAD

**Issue by gaugau3000** [freqtrade/freqtrade#2011]
## Step 3: Describe the problem:

1/The bot place a limit buy order. (unfilledtimeout of the buy = 5 minutes)

> Found open order for Trade(id=17, pair=TFUEL/USDT, amount=26315.78947368, open_rate=0.00760000, open_since=just now)

2/The buy order is partially filled and cancel do to timeout

> Partial buy order timeout for Trade(id=17, pair=TFUEL/USDT, amount=10.20000000, open_rate=0.00760000, open_since=5 minutes ago).

3/the amount is too slow to place a stoploss

> 2019-07-11 07:23:03,825 - freqtrade.freqtradebot - WARNING - Unable to place a stoploss order on exchange: Could not create stop_loss_limit sell order on market TFUEL/USDT.Tried to sell amount 10.2 at rate 0.00715 (total 0.07293).Message: binance {"code":-1013,"msg":"Filter failure: MIN_NOTIONAL"}

4/ the bot crash

```
 2019-07-11 07:38:02,780 - freqtrade - ERROR - Fatal exception!
 Traceback (most recent call last):
   File "/usr/local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 526, in fetch
     response.raise_for_status()
   File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
     raise HTTPError(http_error_msg, response=self)
 requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.binance.com/api/v3/order
 
 During handling of the above exception, another exception occurred:
 
 Traceback (most recent call last):
   File "/freqtrade/freqtrade/exchange/exchange.py", line 370, in create_order
     amount, rate, params)
   File "/usr/local/lib/python3.7/site-packages/ccxt/binance.py", line 703, in create_order
     response = getattr(self, method)(self.extend(request, params))
   File "/usr/local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 421, in inner
     return entry(_self, **inner_kwargs)
   File "/usr/local/lib/python3.7/site-packages/ccxt/binance.py", line 1226, in request
     response = self.fetch2(path, api, method, params, headers, body)
   File "/usr/local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 446, in fetch2
     return self.fetch(request['url'], request['method'], request['headers'], request['body'])
   File "/usr/local/lib/python3.7/site-packages/ccxt/base/exchange.py", line 538, in fetch
     self.handle_errors(response.status_code, response.reason, url, method, headers, http_response, json_response)
   File "/usr/local/lib/python3.7/site-packages/ccxt/binance.py", line 1219, in handle_errors
     raise exceptions[error](self.id + ' ' + body)
 ccxt.base.errors.InvalidOrder: binance {"code":-1013,"msg":"Filter failure: MIN_NOTIONAL"}
 
 During handling of the above exception, another exception occurred:
 
 Traceback (most recent call last):
   File "/freqtrade/freqtrade/main.py", line 52, in main
     worker.run()
   File "/freqtrade/freqtrade/worker.py", line 72, in run
     state = self._worker(old_state=state)
   File "/freqtrade/freqtrade/worker.py", line 110, in _worker
     self._throttle(func=self._process, min_secs=throttle_secs)
   File "/freqtrade/freqtrade/worker.py", line 123, in _throttle
     result = func(*args, **kwargs)
   File "/freqtrade/freqtrade/worker.py", line 133, in _process
     state_changed = self.freqtrade.process()
   File "/freqtrade/freqtrade/freqtradebot.py", line 136, in process
     state_changed |= self.process_maybe_execute_sell(trade)
   File "/freqtrade/freqtrade/freqtradebot.py", line 459, in process_maybe_execute_sell
     result = self.handle_trade(trade)
   File "/freqtrade/freqtrade/freqtradebot.py", line 596, in handle_trade
     if self.check_sell(trade, sell_rate, buy, sell):
   File "/freqtrade/freqtrade/freqtradebot.py", line 711, in check_sell
     self.execute_sell(trade, sell_rate, should_sell.sell_type)
   File "/freqtrade/freqtrade/freqtradebot.py", line 852, in execute_sell
     time_in_force=self.strategy.order_time_in_force['sell']
   File "/freqtrade/freqtrade/exchange/exchange.py", line 412, in sell
     return self.create_order(pair, ordertype, 'sell', amount, rate, params)
   File "/freqtrade/freqtrade/exchange/exchange.py", line 379, in create_order
     f'Could not create {ordertype} {side} order on market {pair}.'
 TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
```

Excepted behavior :
for example no matter the bot don't succeed place a stoploss on MIN_NOTIONAL just close the trade in the database.

Maybe i can prevent this behavior with FOK (Full Or Kill)

Have a nice day.

**Comment by xmatthias** [comment]
yes, FOK will prevent this (also IOK - immediately or kill should).

I think this is actually a partial duplicate of #1858. While it didn't crash in that instance, the half-filled orders are a problem we need to adress.

**Comment by xmatthias** [comment]
I've created #2157 now, which will close this issue.

It will fix the case where the bot will crash (it was due to a unfortunate calculation in an exception handler - introduced by supporting market orders).
The new version will however leave the trade open.
Automatically closing the trade is not that easy, since the error (`InvalidOrder`) is the same, if the problem is MIN_NOTIONAL or something else (which may be temporary).

I wouldn't want to start closing trades because of temporary exchange problems - since that can lead to 100% losses (coin goes down - you don't know because you only look at the bot).

We'll need to deal with half-open orders seperately - as part of #1858 most likely, since that issue is limited to this half-bought case.
